service/history/timerQueueProcessorBase.go

```diff
@@ -578,6 +585,35 @@ func (t *timerQueueProcessorBase) processDeleteHistoryEvent(task *persistence.Ti
 	return t.deleteWorkflowVisibility(task)
 }
 
+func (t *timerQueueProcessorBase) processArchiveHistoryEvent(task *persistence.TimerTaskInfo) (retError error) {
+
+	context, release, err := t.cache.getOrCreateWorkflowExecution(t.getDomainIDAndWorkflowExecution(task))
+	if err != nil {
+		return err
+	}
+	defer func() { release(retError) }()
+
+	msBuilder, err := loadMutableStateForTimerTask(context, task, t.metricsClient, t.logger)
+	if err != nil {
+		return err
+	} else if msBuilder == nil || msBuilder.IsWorkflowExecutionRunning() {
+		// this can happen if workflow is reset.
+		return nil
+	}
+	ok, err := verifyTaskVersion(t.shard, t.logger, task.DomainID, msBuilder.GetLastWriteVersion(), task.Version, task)
+	if err != nil {
+		return err
+	} else if !ok {
+		return nil
+	}
+	return t.historyService.archivalClient.Archive(&sysworkflow.ArchiveRequest{
+		DomainID:         task.DomainID,
```

Comment:
Looks like you are not even using TargetDomainID, TargetWorkflowID and TargetRunID during processing of the task.  You can safely remove them.